### PR TITLE
feat(filter): implement Responses API SSE parser (#501)

### LINKED ISSUE #7: Responses SSE parser: chunk reassembly and event parsing
## Parent

Part of praxis-proxy/ai#101

## Scope

Pure SSE parsing, no business logic:

- Parse Responses SSE events with arbitrary chunk boundaries
- Buffer partial SSE lines across chunks (handle CRLF, LF, partial lines)
- Parse `data:` lines into typed Responses API event structs
- Handle `[DONE]` sentinel and provider-specific termination
- Fail on malformed event JSON
- Enforce limits:
  - `max_stream_buffer_bytes`
  - `max_stream_events`
  - `stream_buffer_timeout_ms`
- Independently testable without a proxy server

## Acceptance criteria

- Split SSE chunks are parsed correctly
- CRLF, LF, partial lines handled
- Malformed event JSON fails closed
- Buffer byte limit is enforced
- Event count limit is enforced
- Timeout is enforced
- Tests cover CRLF, LF, partial lines, malformed data, and large streams

## Reference

- [Proposal: Responses API Filters](docs/proposals/00354_responses-api-filters.md) — Filter 6
- Related: praxis-proxy/ai#81 (generic SSE inspection), praxis-proxy/ai#89
- Part of praxis-proxy/ai#93

### PR DESCRIPTION
**Pull request by leseb**
## Summary

- Implements a three-layer SSE parser for OpenAI Responses API streaming events behind the `ai-inference` feature gate
- **`SseFrameParser`** (frame.rs): byte-level SSE frame reassembly with LF/CRLF/CR handling, buffer overflow protection, and event count budgeting
- **`ResponsesEvent`** (event.rs): typed enum with 26 variants covering all Responses API event types, JSON cross-validation of `event:` vs `data.type`, and dual reasoning event aliasing for OpenResponses compatibility
- **`ResponsesSseParser`** (parser.rs): orchestrator wiring frame parsing to typed events with timeout enforcement, event count limits, completion state tracking, and fail-closed `validate_complete()` semantics
- 131 unit tests covering chunk splitting, CRLF edge cases, buffer overflow, event limits, terminal event detection, and full stream lifecycle

The flow:

```
raw bytes  ──→  frame.rs (accumulates, emits SseFrame)
                    ──→  event.rs (parses, returns ResponsesEvent)
                              ──→  parser.rs (enforces limits, tracks completion)
```

Foundation for praxis-proxy/ai#25 (stream_events filter) and praxis-proxy/ai#6 (stateful orchestration).

## Test plan

- [x] `cargo test -p praxis-proxy-filter --features praxis-proxy-filter/ai-inference` — 131 tests pass
- [x] `make lint` — clippy, fmt, lint-deps all clean
- [x] `make build` — workspace builds cleanly
- [x] Codex xhigh review (gpt-5.5) — 0 findings

Closes praxis-proxy/ai#7

### CODE REVIEW

File: filter/src/builtins/http/ai/openai/sse/responses/parser.rs [hunk 1]
```diff
@@ -0,0 +1,464 @@
...
+            return Err(SseParseError::Timeout {
+                elapsed,
+                limit: self.timeout,
+            });
+        }
+
+        Ok(())
+    }
+
+    /// Record whether an event signals stream completion.
```
[the remark this answers is not in the document]
Comment (leseb) [reply filter/src/builtins/http/ai/openai/sse/responses/parser.rs:107]: https://github.com/praxis-proxy/praxis/pull/721/commits/d888ae680b94388471ea9df0b8c40e2818c1ec5b

File: filter/src/builtins/http/ai/openai/sse/frame.rs [hunk 2]
```diff
@@ -0,0 +1,552 @@
...
+
+        Ok(())
+    }
+
+    /// Process a completed line, emitting a frame on blank lines.
+    #[expect(clippy::indexing_slicing, reason = "colon_pos from position() guarantees bounds")]
+    #[expect(clippy::too_many_lines, reason = "linear SSE field processing")]
+    fn process_line(&mut self) -> Option<SseFrame> {
+        if self.line_buf.is_empty() {
+            let frame = self.has_data.then(|| SseFrame {
```
[the remark this answers is not in the document]
Comment (leseb) [reply filter/src/builtins/http/ai/openai/sse/frame.rs:163]: I ran cargo clippy -p praxis-proxy-filter --all-targets -- -D warnings; it passes. Local clippy.toml sets too-many-lines-threshold = 30, and process_line is over that threshold, so the #[expect] is fulfilled.

File: filter/src/builtins/http/ai/openai/sse/frame.rs [hunk 3]
```diff
@@ -0,0 +1,552 @@
...
+                    }
+                }
+            } else {
+                self.line_buf.push(b);
+            }
+
+            self.scratch_bytes = self.buffered_bytes();
+            if self.scratch_bytes > self.max_buffer_bytes {
+                return Err(SseParseError::BufferOverflow {
+                    buffered_bytes: self.scratch_bytes,
```
[the remark this answers is not in the document]
Comment (leseb) [reply filter/src/builtins/http/ai/openai/sse/frame.rs]: https://github.com/praxis-proxy/praxis/pull/721/commits/ab9bee82da012a881879ebdcd0cc3bc23b6b5a25

**Review by franciscojavierarceo — APPROVED** [review]
